build.gradle

```diff
@@ -409,9 +413,17 @@ bootstrap.dependsOn installTestGems
 runIntegrationTests.shouldRunAfter tasks.getByPath(":logstash-core:test")
 check.dependsOn runIntegrationTests
 
+String artifactsVersionApi = "https://artifacts-api.elastic.co/v1/versions/"
+def dlVersions = new JsonSlurper().parseText(artifactsVersionApi.toURL().text)
+// the version string can be either '7.0.0' or '7.0.0-alpha1', i.e. with the qualifier.
+// in the normal PR type builds it is plain '7.0.0'
+// in the build invoked by the release manager it is '7.0.0-alpha1' etc.
+// the artifacts-api will return JSON like this: `{"versions":["5.6.13-SNAPSHOT","6.4.3-SNAPSHOT","6.5.0-SNAPSHOT","6.6.0-SNAPSHOT","7.0.0-alpha1-SNAPSHOT"]}`
+String qualifiedVersion = dlVersions['versions'].grep(~/^${version}.*/)[0]
```

Comment:
should this be validated?

Reply:
I tested the line in a Groovy online REPL. If there is no match then the `qualifiedVersion` string will be `"NULL"`.

If anything I should swap some lines around and use `qualifiedVersion` more.

If validation is necessary (not saying it is not), it should be done in the `task downloadEs(type: Download)` task.

Reply:
I guess a doFirst closure that `throw new GradleException("could not find the current artifact from the artifact-api for version: ${version}")` if `qualifiedVersion` is "NULL"

Reply:
In the groovy REPL `.toURL().text` throws an exception for an incorrect url `https://artifacts-api.elastic.co/v2/versions/`, so I guess that would halt the script in that case.
If the result is not a serialised JSON object then either the JSON parser throws an exception or the next line complains about a missing property on object `dlVersions` exception.

I think those are covered. I'll add the doFirst block.

Reply:
Retested the full groovy block in the online repl incl. the `.toURL().text` with various "chaos-monkey" like inputs.